tree/dataframe/src/RLoopManager.cxx

```diff
@@ -805,18 +805,25 @@ void RLoopManager::CleanUpTask(TTreeReader *r, unsigned int slot)
 /// This method also clears the contents of GetCodeToJit().
 void RLoopManager::Jit()
 {
-   {
+   auto noCodeToJit = []() {
```

Comment:
The lock guard is redundant in the second invocation and thus could be extracted/used only for the first invocation. (Note that a thread asking for the write lock while holding the read lock is fully supported - read locks are suspended during the request for the write locks).